map/src/components/map.tsx

```diff
@@ -366,6 +371,19 @@ class MapComponent extends React.Component<Props, {}> {
     }
   };
 
+  private centerToGeolocation = () => {
+    navigator.geolocation.getCurrentPosition(position => {
+      const pos = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude,
+      };
+      if (!this.map) {
+        return;
+      }
+      this.map.map.setCenter(pos);
```

Comment:
When we center on location we should run the "Update results for this area" function as well.

Reply:
Agreed, we should perhaps also modify the zoom level

Reply:
I got stuck at updating results. Simply `this.setResults()` doesn't work. Because, after I set the map's center it doesn't change the `nextResults`  immediately. It needs to wait for cluster calculation, then updates the results. I couldn't find a proper way to do it. 

I've just tried to set a timeout for some milliseconds then only after calling `updateResults`, it works but it really is a poor solution.

I'm looking forward to hear your ideas because this problem will cause the same issue when we're implementing the follow-up feature #288 

The event `MarkerClusterer - 'clusteringend'` is responsible for setting nextResults after the bounds are changed. As shown below.
https://github.com/reach4help/reach4help/blob/78b6ee21901f9de72848c692e53a3ae58f66e083/map/src/components/map.tsx#L292-L300

Reply:
for this, i would introduce a boolean property called `updateResultsOnNextClusteringChange` that you set to true before changing the map coordinates, then check that value in the `clusteringend` event listener, and if true, set it to false and call `this.setResults` :)

Reply:
Eureka!!!